flow/connectors/postgres/postgres.go

```diff
@@ -771,11 +772,48 @@ func (c *PostgresConnector) GetTableSchema(
 	return res, nil
 }
 
+func (c *PostgresConnector) GetSelectedColumns(
+	ctx context.Context,
+	sourceTable *utils.SchemaTable,
+	excludedColumns []string,
+) ([]string, error) {
+	// Given the list of excluded columns,
+	// we need to get the list of columns that are not excluded
+	getColumnsSQL := "SELECT attname AS column_name, format_type(atttypid, atttypmod) AS data_type, attnotnull AS is_nullable FROM pg_attribute WHERE attrelid = '%s.%s'::regclass AND attnum > 0 AND NOT attisdropped"
+	rows, err := c.conn.Query(ctx, fmt.Sprintf(getColumnsSQL, sourceTable.Schema, sourceTable.Table))
+	if err != nil {
+		slog.Error("error getting selected columns",
+			slog.Any("error", err),
+			slog.String("table", sourceTable.String()),
+			slog.Any("excludedColumns", excludedColumns))
+		return nil, fmt.Errorf("error getting selected columns for table %s: %w", sourceTable, err)
+	}
+
+	columns := make([]string, 0)
+	for rows.Next() {
+		var columnName, dataType string
+		var isNullable bool
+		if err := rows.Scan(&columnName, &dataType, &isNullable); err != nil {
+			return nil, fmt.Errorf("error scanning column while getting selected columns: %w", err)
+		}
+
+		for _, excludedColumn := range excludedColumns {
+			if columnName == excludedColumn {
+				continue
+			}
+			columns = append(columns, QuoteIdentifier(columnName))
+		}
+	}
+
+	return columns, nil
+}
+
 func (c *PostgresConnector) getTableSchemaForTable(
 	ctx context.Context,
 	env map[string]string,
 	tableName string,
 	system protos.TypeSystem,
+	excludedColumns []string,
```

Comment:
meta: column exclusion is now applied by the columns missing from the schema entirely instead of them being present and filtered at several places
need to test this, I think you'll have schema changes detecting excluded columns and trying to add them

Reply:
Functionally tested and it works.  Looks like we take care of excluded columns in schema changes [here](https://github.com/PeerDB-io/peerdb/blob/6e5ed879ec653299729cc8257f76270840fb11db/flow/connectors/postgres/cdc.go#L934-L935)